src/concurrency/genmc/intercept.rs

```diff
@@ -1,16 +1,136 @@
+use rustc_middle::throw_unsup_format;
 use tracing::debug;
 
 use crate::concurrency::thread::EvalContextExt as _;
 use crate::{
-    BlockReason, InterpResult, MachineCallback, MiriInterpCx, OpTy, UnblockKind, VisitProvenance,
-    VisitWith, callback, interp_ok,
+    BlockReason, InterpResult, MachineCallback, MiriInterpCx, OpTy, Scalar, UnblockKind,
+    VisitProvenance, VisitWith, callback, interp_ok, throw_ub_format,
 };
 
 // Handling of code intercepted by Miri in GenMC mode, such as assume statement or `std::sync::Mutex`.
 
-/// Other functionality not directly related to event handling
 impl<'tcx> EvalContextExt<'tcx> for crate::MiriInterpCx<'tcx> {}
 pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
+    /// Given a `ty::Instance<'tcx>`, do any required special handling.
+    /// Returns true if this `instance` should be skipped (i.e., no MIR should be executed for it).
+    fn check_genmc_intercept_function(
```

Comment:
```suggestion
    fn genmc_intercept_function(
```
That should be clear enough.